src/zope/configuration/tests/test_config.py

```diff
@@ -88,31 +90,28 @@ def test_resolve_bad_dotted_import(self):
             c.resolve('zope.configuration.nosuch.noreally')
         self.assertIn('ImportError', str(exc.exception))
 
+    def _cleanup_bad_imports(self):
+        import sys
+
+        for name in ('zope.configuration.tests.victim',
+                     'zope.configuration.tests.bad'):
+            sys.modules.pop(name, None)
+
     def test_resolve_bad_sub_last_import(self):
         #Import error caused by a bad sub import inside the referenced
         #dotted name. Here we keep the standard traceback.
-        import sys
         c = self._makeOne()
         with self.assertRaises(ImportError):
             c.resolve('zope.configuration.tests.victim')
-        #Cleanup:
-        for name in ('zope.configuration.tests.victim',
-                     'zope.configuration.tests.bad'):
-           if name in sys.modules:
-               del sys.modules[name]
+        self._cleanup_bad_imports()
```

Comment:
Why not move this (and all other) calls to the `tearDown`, so it gets run even if the test fails?

Reply:
Good point, will do.